infrastructure/src/main/java/org/wso2/carbon/testgrid/infrastructure/aws/AWSDeployer.java

```diff
@@ -0,0 +1,190 @@
+* Version 2.0 (the "License"); you may not use this file except
+* in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.carbon.testgrid.infrastructure.aws;
+
+import com.amazonaws.auth.EnvironmentVariableCredentialsProvider;
+import com.amazonaws.services.cloudformation.AmazonCloudFormation;
+import com.amazonaws.services.cloudformation.AmazonCloudFormationClientBuilder;
+import com.amazonaws.services.cloudformation.model.CreateStackRequest;
+import com.amazonaws.services.cloudformation.model.DeleteStackRequest;
+import com.amazonaws.services.cloudformation.model.DescribeStacksRequest;
+import com.amazonaws.services.cloudformation.model.DescribeStacksResult;
+import com.amazonaws.services.cloudformation.model.Output;
+import com.amazonaws.services.cloudformation.model.Stack;
+import com.amazonaws.services.cloudformation.model.StackStatus;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.testgrid.common.Deployment;
+import org.wso2.carbon.testgrid.common.Host;
+import org.wso2.carbon.testgrid.common.Infrastructure;
+import org.wso2.carbon.testgrid.common.Script;
+import org.wso2.carbon.testgrid.common.exception.TestGridInfrastructureException;
+import org.wso2.carbon.testgrid.utils.EnvVariableUtil;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class is responsible for creating the AWS infrastructure.
+ */
+public class AWSDeployer {
+
+    private Infrastructure infra;
+    private static final Log log = LogFactory.getLog(AWSDeployer.class);
+
+    /**
+     * This constructor creates AWS deployer object and validate AWS related environment variables are present.
+     *
+     * @param awsKeyVariableName    Environment variable name for AWS ACCESS KEY.
+     * @param awsSecretVariableName Environment variable name for AWS SECRET KEY.
+     * @throws TestGridInfrastructureException Throws exception when environment variables are not set.
+     */
+    public AWSDeployer(String awsKeyVariableName, String awsSecretVariableName) throws TestGridInfrastructureException {
+        String awsIdentity = EnvVariableUtil.readEnvironmentVariable(awsKeyVariableName);
+        String awsSecret = EnvVariableUtil.readEnvironmentVariable(awsSecretVariableName);
+        if ((awsIdentity == null) || (awsSecret == null)) {
+            String errorMessage = "AWS Credentials must be set as environment variables";
+            throw new TestGridInfrastructureException(errorMessage);
```

Comment:
`throw new TestGridInfrastructureException("AWS Credentials must be set as environment variables");` would be much better (since error message is not used else where)